packages/server-core/src/user/user/user.resolvers.ts

```diff
@@ -63,7 +64,13 @@ export const userDataResolver = resolve<UserType, HookContext>({
 })
 
 export const userPatchResolver = resolve<UserType, HookContext>({
-  updatedAt: getDateTimeSql
+  updatedAt: getDateTimeSql,
+  deactivatedAt: async (deactivatedAt) => {
+    if (deactivatedAt) {
+      return getDateTimeSql()
+    }
+    return undefined
+  }
```

Comment:
This check should be based on `isDeactivated` prop

```suggestion
  deactivatedAt: async (_, userData) => {
    if (userData.isDeactivated) {
      return getDateTimeSql()
    }
    return undefined
  }
```

Reply:
addressed ty!